common/http/platform/beast/http_resumer.cpp

```diff
@@ -0,0 +1,454 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/http.hpp>
+
+#include <algorithm>
+
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl/verify_mode.hpp>
+#include <boost/asio.hpp>
+
+#include <common/common.hpp>
+
+namespace mender {
+namespace http {
+
+namespace common = mender::common;
+
+// Represents the parts of a Content-Range HTTP header
+// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range
+struct RangeHeader {
+	long long int range_start;
+	long long int range_end;
+	long long int size;
+};
+using ExpectedRangeHeader = expected::expected<RangeHeader, error::Error>;
+ExpectedRangeHeader ParseRangeHeader(string resp);
+
+ExpectedRangeHeader ParseRangeHeader(string header) {
+	RangeHeader range_header {.range_start = 0, .range_end = 0, .size = 0};
```

Comment:
I'm a bit on the fence. Your version is obviously simpler, but I don't find it *that* bad. Long, but not difficult.

Of course, if I wasn't worried about having to rip out regex later because of size constraints on embedded devices, it would be a no-brainer: yours is clearly better. But the regex argument is really just a hunch, so I can't use it as a strong argument.

How about this suggestion: Make a comment in the code that a non-regex version exists in this PR (or some other suitable branch), and switch to regex. Then our code will be nice and clean, but at least we can get the non-regex version back if we need to, without having to rewrite it.